deepseek-model-provider/tool.gpt

```diff
@@ -9,3 +9,10 @@ Credential: ../model-provider-credential as deepseek-model-provider
 ---
 !metadata:*:icon
 /admin/assets/deepseek-logo.svg
+
+---
+Name: validate
+Description: Validate the DeepSeek API key
+Metadata: envVars: OBOT_DEEPSEEK_MODEL_PROVIDER_API_KEY
+
+#!${GPTSCRIPT_TOOL_DIR}/validate/bin/gptscript-go-tool
```

Comment:
I don't think this works in general. For example, if you run Obot outside of the Docker container, then I believe this will fail. Specifically, it won't be able to find the `gptscript-go-tool` binary. The reason: GPTScript will only build the `main.go` file at the root of the repo; it won't look for nested `main.go` files. It work in Obot when you set the `OBOT_REGISTRY_URL` to a local directory and you have already run `make build` in the tools repo, then this will work.

I think the approach would be to have arguments: `server` and `validate` to the `main.go` file and run the respective code from there.

Reply:
makes sense